plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergLocalConcurrentWrites.java

```diff
@@ -1285,6 +1285,122 @@ private void runOptimizeDuringWriteOperations(boolean useSmallFiles)
         }
     }
 
+    // Repeat test with invocationCount for better test coverage, since the tested aspect is inherently non-deterministic.
+    @RepeatedTest(3)
+    void testConcurrentNonOverlappingOptimize()
+            throws Exception
+    {
+        testConcurrentNonOverlappingOptimize(true);
+        testConcurrentNonOverlappingOptimize(false);
+    }
+
+    private void testConcurrentNonOverlappingOptimize(boolean partitioned)
```

Comment:
Why does it work in non-partitioned tables?

Reply:
Naming error